app/controllers/sources_controller.rb

```diff
@@ -94,4 +93,15 @@ def source_params
   def load_source_set
     @source_set = SourceSet.friendly.find(params[:source_set_id])
   end
+
+  def parse_dpla_item(dpla_item)
+    return unless dpla_item.present?
+    @digital_resource_url = dpla_item.try(:url)
+    @provider_name = dpla_item.try(:provider).try(:name)
+    data_provider = dpla_item.try(:source)
+    intermediate_provider = dpla_item.try(:intermediate_provider)
+    @contributing_institution = 
+      [data_provider, intermediate_provider].compact.join('; ')
+    @title = Array(dpla_item.try(:title)).flatten.first
+  end
```

Comment:
Let's make a new issue to refactor this. I think there's a lot we could do to make this somewhat cleaner but at first glance we may want to look at diverging from `DPLibrary`'s implementation so we can get what we need out of the object.